src/Flex.php

```diff
@@ -757,18 +757,33 @@ public function fetchRecipes(array $operations): array
             }
 
             $noRecipe = !isset($manifests[$name]) || (isset($manifests[$name]['not_installable']) && $manifests[$name]['not_installable']);
-            if ($noRecipe && 'symfony-bundle' === $package->getType()) {
-                $manifest = [];
-                $bundle = new SymfonyBundle($this->composer, $package, $job);
-                if (null === $devPackages) {
-                    $devPackages = array_column($this->composer->getLocker()->getLockData()['packages-dev'], 'name');
+            if ($noRecipe) {
+                $bundles = [];
+
+                $extra = $package->getExtra();
+                if (isset($extra['symfony']['bundles'])) {
+                    $extraBundles = (array) $extra['symfony']['bundles'];
+                    foreach ($extraBundles as $bundleClass => $envs) {
+                        $bundles[$bundleClass] = (array) $envs;
+                    }
                 }
-                $envs = \in_array($name, $devPackages) ? ['dev', 'test'] : ['all'];
-                foreach ($bundle->getClassNames() as $class) {
-                    $manifest['manifest']['bundles'][$class] = $envs;
+
+                if (0 === \count($bundles)) {
```

Comment:
this should be the "else" case from the previous "if": it's legit to say there are no bundles in this lib while autodiscovery might find some otherwise, isn't it?

Reply:
Currently empty `symfony.bundles` should not disable the autodiscovery. With the change you propose, it will disable it. 
So it is more the question of how we want it to behave? :)

Reply:
It should skip then - we should never remove power from users.

Reply:
But being able to disable the autodiscovery with setting an empty `symfony.bundles` might be a good option!

Reply:
https://github.com/symfony/flex/pull/612/commits/b5aebef599fd8d0eb868a9de27d9fdc8ef1b3929

Reply:
Which is what I'm saying :)